### PR TITLE
AA compliance with nonprod verification page

### PR DESCRIPTION
**Pull request by ATBull81**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [ ] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/story/show/186426731

# A brief description of the changes

Current behavior: Session expiring and re-login modals were not navigable via keyboard, along with the non-prod radio button options

New behavior: You can now navigate throughout each modal and the non-prod radio buttons. On top of this, translations are added.

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable that is used to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name: CR97_IS_ENABLED

- [ ] DC
- [x] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

Cucumber tests will be added via ticket at the end of the workflow (https://www.pivotaltracker.com/story/show/186580246)

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

This PR is being flagged as a failure by AppScan codesweep - and it is a false positive.

The code in question is [app/views/devise/sessions/_session_expiration_warning.html.erb](https://github.com/ideacrew/enroll/pull/3368/files#diff-48ccd5195e0545922084d81e1e2ec61086d70c0bc9ccaeb2ee2a04294f101a0b) which reads:

```erb
<div class="modal-footer">
  <%= link_to l10n(".logout"), destroy_user_session_path, method: "delete", class: "btn btn-default btn-small", tabindex: "0" %>
  <span tabindex="0", onkeydown="handleButtonKeyDown(event, 'stay-logged-in')", id="stay-logged-in" class="btn btn-primary  mtz btn-small" data-dismiss="modal"><%= l10n("faa.stay_logged_in") %></span>
</div>
````
CodeSweep is catching all uses of the Rails link_to helper, whether or not it is receiving dynamically generated input (see the screenshot where all arguments are hardcoded strings).  This is likely because link_to [does not do automatic string escaping](https://www.stackhawk.com/blog/rails-xss-examples-and-prevention/#:~:text=The%20above%20code%20is%20XSS%20vulnerable.%20Why%3F%20The%20user%20can%20provide%20JavaScript%20code%20as%20the%20portfolio%20URL.%20And%20since%20link_to%20wasn%E2%80%99t%20designed%20to%20be%20used%20with%20user%2Dgenerated%20input%2C%20it%20doesn%E2%80%99t%20automatically%20do%20string%20escaping.) since it was not designed to be used with user-generated input. The code in question has two pieces of dynamically generated input, both of which are known to be safe:

- l10n(".logout") - This is a call to the translation helper method that reads hardcoded string values from a set of files checked in to the codebase.  There is no user-facing aspect of this code; it is managed by our standard version control processes.

- destroy_user_session_path - this is a conventional path helper method included in the Devise authentication gem.  It also is not generated using user input.

Screenshot 2023-12-11 at 2 55 47 PM

### CODE REVIEW

**Review by yosefserkez — APPROVED** [review]

**Review by mdkaraman — APPROVED** [review]
Code changes look good.  I am approving this PR strictly as a second peer reviewer.  Since there is a new false positive being flagged here, we need a security review from Desire and the additional review/approvals outlined here: https://github.com/ideacrew/enroll/blob/trunk/appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

**Review by desireB — APPROVED** [review]
I have confirm that the link_to helper is not:
- receiving input generated by the user
- the translation helper method is fine
I read also [does not do automatic string escaping](https://www.stackhawk.com/blog/rails-xss-examples-and-prevention/#:~:text=The%20above%20code%20is%20XSS%20vulnerable.%20Why%3F%20The%20user%20can%20provide%20JavaScript%20code%20as%20the%20portfolio%20URL.%20And%20since%20link_to%20wasn%E2%80%99t%20designed%20to%20be%20used%20with%20user%2Dgenerated%20input%2C%20it%20doesn%E2%80%99t%20automatically%20do%20string%20escaping.) 

I confirm that this is false positive and I approve the changes.

**Comment by mdkaraman** [comment]
Confirming that Angus has approved merge of this PR to trunk.

**Pull request merged**
